### ISSUE TITLE
protect UI API calls with POST/csrf

### ISSUE THREAD

**Issue by joefutrelle** [WHOIGit/ifcbdb#299]
Right now we have GET API calls that users can call from non-browser clients, and some of them are expensive. For endpoints that are just used for the UI, they should be POSTs that require a csrf token.

**Comment by joefutrelle** [comment]
TODO: idedntify relevant endpoints

**Comment by joefutrelle** [comment]
Agreed that this is not necessary for all endpoints and not essential for many of them. So I will identify the ones to target and we can just do those for now. My criteria are

1. Does the endpoint perform an expensive and/or asynchronous operation (e.g., mosaic computation)
1. Does the endpoint expose internal information like database IDs that callers should not rely on

I think this will be a reasonably small subset of the API endpoints. In the next round we would protect all API endpoints that aren't appropriate to be called from an external script but really only support the UI.

**Comment by mike-kaimika** [comment]
@joefutrelle I finished up the conversion of all the items that start with /api, minus about 4 of them that I think should remain as GETS (image links mostly). All changes are in the below PR. As mentioned in Slack, I've done the conversion AND tried to locate and test all instances where those methods are called, and everything looks good as far as I can tell

**Updated Files**
https://github.com/WHOIGit/ifcbdb/pull/326
